packages/vite/src/node/plugins/define.ts

```diff
@@ -93,6 +93,12 @@ export function definePlugin(config: ResolvedConfig): Plugin {
         result.map = s.generateMap({ hires: true })
       }
       return result
+    },
+    transformIndexHtml(html: string) {
+      if (config.defineHtml !== true) return html
+      return html.replace(pattern, (_, match) => {
+        return '' + replacements[match]
```

Comment:
**issue (blocking):** Seems that `replacements[match]` could theoretically result in `undefined`  
`'' + undefined` == `'undefined'`
I don't think this is wanted behavior, isn't it?

Reply:
Also you could shorten line 99 - 101 by using a one-liner instead of an inner return body

Reply:
Thanks for your reply, this code fragment is copied from [here](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/define.ts#L70) (They are focusing on one highly similar task), so I shared the code.

And I added detailed descriptions, If you agree with the initial purpose of the PR, I will change the code style, otherwise please feel free to close it.